README.md

```diff
@@ -1,13 +1,113 @@
 # Node.js API for MaxMind minFraud Score, Insights, and Factors
 
+## Description
+This package provides an API for the [MaxMind minFraud Score, Insights, and
+Factors web services](https://dev.maxmind.com/minfraud/).
+
+## Requirements
+
+MaxMind has tested this API with Node.js versions 8 and 10.  We aim to support
+active LTS versions of Node.js, as well as the latest stable release.
+
+## Installation
+
+```
+yarn add @maxmind/minfraud-api-node
+```
+
+If you are not able to use `yarn`, you may also use `npm`:
+
+```
+npm install @maxmind/minfraud-api-node
+```
+
+## API Documentation
+
+Documentation for this API can be found [here](https://maxmind.github.io/minfraud-api-node/)
+
+## Usage
+To use this API, first create a new `Client` object. The constructor
+takes your MaxMind account ID and license key. For example:
+
+```js
+const client = new minFraud.Client("1234", "LICENSEKEY");
+```
+
+Then create a new `Transaction` object. This represents the transaction that
+you are sending to minFraud. Each transaction property is instantiated by creating
+a new instance of each property's class.  For example:
+
+```js
+const transaction = new minFraud.Transaction({
+  device: new minFraud.Device({
+    ipAddress: '8.8.8.8',
+  }),
+  email: new minFraud.Email({
+    address: 'foo@bar.com',
+    domain: 'bar.com',
+  }),
+})
+```
+
+If Transaction instantiation fails, an `ArgumentError` is thrown. This is usually
+due to invalid property values, a missing Device object, or a Device object
+ with either an invalid or missing IP Address.
+
+After creating the Transaction object, you can send a Score, Insights, or Factors
+request, which returns a Promise that contains the corresponding model:
+
+```js
+// minFraud Score
+client.score(transaction).then(scoreResponse => ...);
+
+// minFraud Insights
+client.insights(transaction).then(insightsResponse => ...);
+
+// minFraud Factors
+client.factors(transaction).then(factorsResponse => ...);
+
+```
+
+If the request fails, an error object will be returned in the catch in the form
+of:
+
+```js
+{
+  code: string
+  error: string
+}
+```
+
 ## Errors and Exceptions
 
 Thrown by the request and transaction models:
-* `ArgumentError` - Thrown when invalid data is passed to the model constructor.
+* `ArgumentError` - Thrown when invalid data is passed to the Transaction 
+and Transaction property constructors.
 
-## Example
+In addition to the [response errors](https://dev.maxmind.com/minfraud/#Errors)
+returned by the web API, we also return: 
 
+```js
+{
+  code: 'SERVER_ERROR',
+  error: 'Some message',
```

Comment:
Oops, I missed this in the previous review, is `error` really "Some message" ?  Please correct if needed

Reply:
It's a stand-in for some dynamically generated message (which could change in the future).  Any suggestions to improve this?

Reply:
What about `<string>` or `...`?